pkg/controllers/namespace/namespaceController.go

```diff
@@ -0,0 +1,268 @@
+	uruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/pkg/api/v1"
+	rest "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	"strings"
+	"time"
+)
+
+// NamespaceController Implements Controller interface
+// Responsible for monitoring kubernetes namespaces and
+// syncing them to Calico datastore.
+type NamespaceController struct {
+	indexer        cache.Indexer
+	informer       cache.Controller
+	calicoObjCache calicocache.ResourceCache
+	calicoClient   *client.Client
+}
+
+// NewNamespaceController Constructor for NamespaceController
+func NewNamespaceController(k8sConfig *rest.Config) controller.Controller {
+
+	clientset, err := kubernetes.NewForConfig(k8sConfig)
+	if err != nil {
+		panic(err.Error())
+	}
+
+	cconfig, err := client.LoadClientConfig("")
+	if err != nil {
+		panic(err)
+	}
+	cclient, err := client.New(*cconfig)
+	if err != nil {
+		panic(err)
+	}
+
+	// Function returns slice of profile objects stored by policy controller
+	// on ETCD datastore.Indentifies controller writen objects by
+	// their naming convention.
+	listFunc := func() ([]interface{}, error) {
+
+		var filteredProfiles []interface{}
+
+		// Get all profile objects from ETCD datastore
+		calicoProfiles, err := cclient.Profiles().List(api.ProfileMetadata{})
+		if err != nil {
+			return filteredProfiles, err
+		}
+
+		// Filter out only objects that are written by policy controller
+		for _, profile := range calicoProfiles.Items {
+
+			profileName := profile.Metadata.Name
+			if strings.HasPrefix(profileName, converter.ProfileNameFormat) {
+				filteredProfiles = append(filteredProfiles, profile)
+			}
+		}
+		glog.Debugf("Found %d profiles in calico ETCD:", len(filteredProfiles))
+		return filteredProfiles, nil
+	}
+
+	// Function returns key of the object in kubernetes format.
+	keyFunc := func(obj interface{}) string {
+		profile := obj.(api.Profile)
+		Key := profile.Metadata.Name
+		return Key
+	}
+
+	cacheArgs := calicocache.ResourceCacheArgs{
+		ListFunc:   listFunc,
+		KeyFunc:    keyFunc,
+		Client:     cclient,
+		ObjectType: "*api.Profile", // Restrict cache to store pointers to calico profile objects only.
+	}
+
+	ccache := calicocache.NewResourceCache(cacheArgs)
+	namespaceConverter := converter.NewNamespaceConverter()
+
+	// create the watcher
+	listWatcher := cache.NewListWatchFromClient(clientset.Core().RESTClient(), "namespaces", "", fields.Everything())
+
+	// Bind the calico cache to kubernetes cache with the help of an informer. This way we make sure that
+	// whenever the kubernetes cache is updated, changes get reflected in calico cache as well.
+	indexer, informer := cache.NewIndexerInformer(listWatcher, &v1.Namespace{}, 0, cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+
+			key, err := cache.MetaNamespaceKeyFunc(obj)
+			glog.Infof("Got ADD event for namespace: %s\n", key)
+
+			if err == nil {
```

Comment:
I think we should instead have an `if err != nil` branch here, so we can log out any errors.